src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -492,24 +487,29 @@ public void restoreFeedbackSessionFromRecycleBin(String feedbackSessionName, Str
      * Gets the expected number of submissions for a feedback session.
      */
     public int getExpectedTotalSubmission(FeedbackSessionAttributes fsa) {
-        List<StudentAttributes> students = studentsLogic.getStudentsForCourse(fsa.getCourseId());
-        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(fsa.getCourseId());
-        List<FeedbackQuestionAttributes> questions =
-                fqLogic.getFeedbackQuestionsForSession(fsa.getFeedbackSessionName(), fsa.getCourseId());
-        List<FeedbackQuestionAttributes> studentQns = fqLogic.getFeedbackQuestionsForStudents(questions);
+        int numOfStudents = studentsLogic.getNumberOfStudentsForCourse(fsa.getCourseId());
+        List<String> instructorEmails = instructorsLogic.getInstructorEmailsForCourse(fsa.getCourseId());
 
         int expectedTotal = 0;
+        if (fqLogic.hasFeedbackQuestionsForStudents(fsa)) {
+            expectedTotal += numOfStudents;
+        }
 
-        if (!studentQns.isEmpty()) {
-            expectedTotal += students.size();
+        // Check a creator and a non-creator once each to prevent unnecessary data read.
+        List<String> creatorEmails = instructorEmails.stream()
+                .filter(fsa::isCreator)
+                .collect(Collectors.toList());
+        if (!creatorEmails.isEmpty()
+                && fqLogic.hasFeedbackQuestionsForInstructors(fsa, true)) {
+            expectedTotal += creatorEmails.size();
```

Comment:
This means a creator instructor will be counted twice?

Reply:
There is a partitioning in the list filtering where `creatorEmails` are creators only (should be singular but who knows in the future) and `nonCreatorEmails` are the rest only.

In hindsight, this is also not needed.